### PR TITLE
Ignore default value for required parameter

### LINKED ISSUE #566: Default value should not be validated for required parameter
Minimal example:
```java
import com.beust.jcommander.IValueValidator;
import com.beust.jcommander.JCommander;
import com.beust.jcommander.Parameter;
import com.beust.jcommander.ParameterException;

public class JC {
	@Parameter(names = "-shift", required = true, validateValueWith = StrictlyPositiveInteger.class)
	private int shift = 1; // FIXME: we must not need to specify a default value compatible with StrictlyPositiveInteger when required is true

	public static void main(String[] args) {
		JCommander.newBuilder()
			.addObject(new JC())
			.build()
			.parse(args);
	}

	public static class StrictlyPositiveInteger implements IValueValidator<Integer> {
		@Override
		public void validate(String name, Integer i) throws ParameterException {
			if (i <= 0) {
				throw new ParameterException("Parameter " + name
					+ " should be strictly positive (found " + i + ")");
			}
		}
	}
}
```
If we remove the default meaningless (because parameter is required) value of `shift` (`private int shift;`), JCommander throws:
```
Exception in thread "main" com.beust.jcommander.ParameterException: Parameter -shift should be strictly positive (found 0)
	at JC$StrictlyPositiveInteger.validate(JC.java:21)
	at JC$StrictlyPositiveInteger.validate(JC.java:17)
	at com.beust.jcommander.ParameterDescription.validateValueParameter(ParameterDescription.java:364)
	at com.beust.jcommander.ParameterDescription.validateValueParameter(ParameterDescription.java:353)
	at com.beust.jcommander.ParameterDescription.validateDefaultValues(ParameterDescription.java:164)
	at com.beust.jcommander.ParameterDescription.init(ParameterDescription.java:157)
	at com.beust.jcommander.ParameterDescription.<init>(ParameterDescription.java:65)
	at com.beust.jcommander.JCommander.addDescription(JCommander.java:631)
	at com.beust.jcommander.JCommander.createDescriptions(JCommander.java:601)
	at com.beust.jcommander.JCommander.parse(JCommander.java:361)
	at com.beust.jcommander.JCommander.parse(JCommander.java:342)
	at JC.main(JC.java:14)
```

See 
https://github.com/cbeust/jcommander/blob/748ddba20a28956332b413eaa99d7e8b5422104d/src/main/java/com/beust/jcommander/ParameterDescription.java#L152-L159

### PR DESCRIPTION
**Pull request by gwenn**
Fix #566

### CODE REVIEW

File: src/test/java/com/beust/jcommander/JCommanderTest.java [hunk 1]
```diff
@@ -1447,4 +1447,29 @@ class Args {
...
+        class Parameters {
+            @Parameter(names = "-v",
+                    required = true,
+                    validateValueWith = StrictlyPositiveInteger.class)
+            private int value;
+        }
+
+        String[] argv = { "-v", "1" };
+        Parameters args = new Parameters();
+        JCommander.newBuilder().addObject(args).build().parse(argv);
```
Comment (mkarg) [inline src/test/java/com/beust/jcommander/JCommanderTest.java:1462]: Looks like a duplicate line. The next line does exactly the same.

**Review by mkarg — CHANGES_REQUESTED** [review]

File: src/test/java/com/beust/jcommander/JCommanderTest.java (hunk 1 above)
Comment (gwenn) [reply src/test/java/com/beust/jcommander/JCommanderTest.java:1462]: Sorry, I've copied / pasted this block:
https://github.com/cbeust/jcommander/blob/748ddba20a28956332b413eaa99d7e8b5422104d/src/test/java/com/beust/jcommander/JCommanderTest.java#L1297-L1311

Comment (mkarg) [reply src/test/java/com/beust/jcommander/JCommanderTest.java:1462]: Can you please update the PR?

**Review by mkarg — CHANGES_REQUESTED** [review]

File: src/test/java/com/beust/jcommander/JCommanderTest.java (hunk 1 above)
Comment (gwenn) [reply src/test/java/com/beust/jcommander/JCommanderTest.java:1462]: See 632dfe0
I've fixed only my lines.
And not https://github.com/cbeust/jcommander/blob/748ddba20a28956332b413eaa99d7e8b5422104d/src/test/java/com/beust/jcommander/JCommanderTest.java#L1308-L1310

**Review by mkarg — APPROVED** [review]

**Comment by mkarg** [comment]
@gwenn Thank you for reporting and fixing this issue! :-)

**Comment by gwenn** [comment]
> @gwenn Thank you for reporting and fixing this issue! :-)

Thank you for helping me !

**Pull request merged**